src/OrchardCore.Modules/Orchard.Autoroute/Services/AutorouteValidator.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Microsoft.Extensions.Localization;
+using Orchard.Autoroute.Model;
+using Orchard.ContentManagement.Records;
+using System;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using YesSql.Core.Services;
+
+namespace Orchard.Autoroute.Services
+{
+    public class AutorouteValidator : IAutorouteValidator
+    {
+        private readonly ISession _session;
+        private readonly IStringLocalizer<AutorouteValidator> T;
+
+        public AutorouteValidator(ISession session, IStringLocalizer<AutorouteValidator> localizer)
+        {
+            _session = session;
+            T = localizer;
+        }
+
+        public async Task ValidateAsync(AutoroutePart autoroute, Action<string, string> reportError)
+        {
+            if (!Regex.IsMatch(autoroute.Path, @"^[^:?#\[\]@!$&'()*+,.;=\s\""\<\>\\\|%]+$"))
```

Comment:
Regex are evil, use a simple array of chars if possible, and Array.Search/IndexOf